server/lib/schema/index.ts

```diff
@@ -289,6 +290,46 @@ const schema = new GraphQLSchema({
           });
         }
       },
+      updateWorkspaceIsPublic: {
+        type: workspaceType,
+        args: {
+          isPublic: { type: GraphQLBoolean },
+          workspaceId: { type: GraphQLString },
+        },
+        resolve: async (_, { isPublic, workspaceId }, context) => {
+          const user = await userFromAuthToken(context.authorization);
+          if (user == null) {
+            throw new Error(
+              "No user found when attempting to toggle workspace visiblity."
```

Comment:
I don't recommend task-specific error names like this. The mutation/task should be obvious in the logs from other info.  I recommend instead making ``checkUser`` or something as one standard method to do authorization.

Reply:
I adopted this style just because it had been used throughout the schema. 

I think factoring this out into a separate `requireUser` is a good idea. My preference is to make this a separate refactor PR, since imposing this style uniformly throughout the schema will involve changing many resolvers not related to the admin controls.

Reply:
yep, makes sense